Ports: Add fribidi

n0toose opened the pull request:
This is a part of a greater effort to port SuperTux.

n0toose commented:
Hang on, seems like I used the lint script incorrectly.

n0toose commented:
Should be fixed now, haven't done this in a bit ^^

ADKaster commented on Ports/fribidi/package.sh:
```diff
@@ -0,0 +1,31 @@
+useconfigure='true'
+use_fresh_config_sub='true'
+archive_hash='7fa16c80c81bd622f7b198d31356da139cc318a63fc7761217af4130903f54a2'
+use_fresh_config_sub='true'
```
you've got `use_fresh_config_sub` twice :o

ADKaster commented on Ports/fribidi/package.sh:
```diff
@@ -0,0 +1,31 @@
+use_fresh_config_sub='true'
+workdir="fribidi-${version}"
+configopts=(
+    "--prefix=${SERENITY_INSTALL_ROOT}/usr/local"
```
--prefix isn't passed to any other meson ports. is it strictly required?

n0toose replied:
The local version worked, but apparently the PR I shipped is slightly problematic in various ways. I will take a look against what other meson ports did and try to clean things up to improve the port's quality.

n0toose commented:
(Forgot the ReadMe.md, will do another force-push)

Other than that, I fixed the issues that were mentioned in the review, used `" "` only when necessary, fixed unused configuration options and cleaned things up.

n0toose commented:
Additional note:

The only other recipe using meson the same way that I do is [`wayland`](https://github.com/SerenityOS/serenity/blob/b37f3c86fd5bd4f8f0ee2be69da331620ad5c211/Ports/wayland/package.sh#L5), I am not sure if I left something out.

timschumi commented on Ports/fribidi/patches/ReadMe.md:
````diff
@@ -0,0 +1,23 @@
+-endif
+```
+
+The inline keyword (C99) is too new for ANSI/C89.
+
+Without this patch, the compiler freaks out because fribidi
+relies on inline functions.
````
Please put these three lines into the commit description (of the commit that this patch is based on) and regenerate the patch and README.

timschumi commented on Ports/fribidi/patches/ReadMe.md:
```diff
@@ -0,0 +1,23 @@
+
+The inline keyword (C99) is too new for ANSI/C89.
+
+Without this patch, the compiler freaks out because fribidi
+relies on inline functions.
```
If this is the case, shouldn't this patch be upstreamable? No need to do that now, just asking.

n0toose replied:
I would upstream it, but I unfortunately do not understand why it exists there in the first place.

n0toose commented:
I plan to work on this next month. I will close the PR for now so as to be able to reopen it again by myself.
